stderr: moveit_core

/usr/bin/ld: cannot find -lEigen3::Eigen
collect2: error: ld returned 1 exit status
make[2]: *** [robot_model/libmoveit_robot_model.so.2.0.0] Error 1
make[1]: *** [robot_model/CMakeFiles/moveit_robot_model.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

Starting >>> moveit_core
--- stderr: moveit_core
CMake Error at /home/ssw/ros2_dashing/install/ament_cmake_gtest/share/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake:50 (add_executable):
  Target "test_robot_model" links to target "Eigen3::Eigen" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  /home/xxx/ros2_dashing/install/ament_cmake_gtest/share/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake:37 (_ament_add_gtest_executable)
  /home/xxx/ros2_dashing/install/ament_cmake_gtest/share/ament_cmake_gtest/cmake/ament_add_gtest.cmake:66 (ament_add_gtest_executable)
  robot_model/CMakeLists.txt:33 (ament_add_gtest)

Starting >>> moveit_core
--- stderr: moveit_core
CMake Error at /home/xxx/ros2_dashing/install/ament_cmake_gtest/share/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake:50 (add_executable):
  Target "test_robot_model" links to target "Eigen3::Eigen" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  /home/xxx/ros2_dashing/install/ament_cmake_gtest/share/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake:37 (_ament_add_gtest_executable)
  /home/xxx/ros2_dashing/install/ament_cmake_gtest/share/ament_cmake_gtest/cmake/ament_add_gtest.cmake:66 (ament_add_gtest_executable)
  robot_model/CMakeLists.txt:33 (ament_add_gtest)

--- a/moveit_core/CMakeLists.txt
+++ b/moveit_core/CMakeLists.txt
@@ -25,6 +25,7 @@ find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(Boost REQUIRED system filesystem date_time thread iostreams)
 
+find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
 find_package(PkgConfig REQUIRED)
 
@@ -129,6 +130,8 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   tf2_geometry_msgs
   trajectory_msgs
   visualization_msgs
+  Eigen3
+  eigen3_cmake_module
   Boost
   OCTOMAP
 )

--- a/moveit_core/package.xml
+++ b/moveit_core/package.xml
@@ -19,6 +19,7 @@
   <buildtool_depend>ament_cmake</buildtool_depend>
   <buildtool_depend>pkg-config</buildtool_depend>
   <buildtool_depend>eigen3_cmake_module</buildtool_depend>
+  <buildtool_export_depend>eigen3_cmake_module</buildtool_export_depend>
 
   <depend>rclcpp</depend>
   <depend>assimp</depend>

--- a/moveit_core/robot_model/CMakeLists.txt
+++ b/moveit_core/robot_model/CMakeLists.txt
@@ -14,17 +14,18 @@ add_library(${MOVEIT_LIB_NAME} SHARED
   src/robot_model.cpp
 )
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-
+ament_target_dependencies(${MOVEIT_LIB_NAME}
+  Eigen3
+  geometric_shapes
+  urdf
+  urdfdom_headers
+  srdfdom
+  visualization_msgs
+)
 target_link_libraries(${MOVEIT_LIB_NAME}
   moveit_profiler
   moveit_exceptions
   moveit_kinematics_base
-  Eigen3::Eigen
-  ${urdf_LIBRARIES}
-  ${srdfdom_LIBRARIES}
-  ${urdfdom_headers_LIBRARIES}
-  ${visualization_msgs_LIBRARIES}
-  ${geometric_shapes_LIBRARIES}
 )
 
 if(BUILD_TESTING)

